lib/streamlit/elements/toast.py

```diff
@@ -84,7 +84,10 @@ def toast(
         """
         toast_proto = ToastProto()
         toast_proto.body = clean_text(validate_text(body))
-        toast_proto.icon = validate_emoji(icon)
+        if icon is not None and icon.startswith(":material"):
+            toast_proto.icon = validate_material_icon(icon)
+        else:
+            toast_proto.icon = validate_emoji(icon)
```

Comment:
Hm, this code snippet appears enough times (here and in `alert.py`) that I think it may be worth generalizing into a helper function, although maybe it's not ideal since the difference in each of the code snippets is the exact proto type that we're setting the `.icon` field for